Dockerfile

```diff
@@ -1,18 +1,26 @@
-FROM rocker/rstudio:3.4.1
+FROM rocker/tidyverse:3.4.2
 
 RUN apt-get update && \
     apt-get -y install python3-pip && \
-    pip3 install --no-cache-dir notebook==5.* 
+    pip3 install --no-cache-dir notebook==5.2 && \
+    apt-get purge && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
+RUN R --quiet -e "install.packages(c('repr', 'IRdisplay', 'evaluate', 'crayon', 'pbdZMQ', 'devtools', 'uuid', 'digest'))"
```

Comment:
you could write this as 

```
install2.r repr IRdisplay evaluate crayon devtools pdbZMQ uuid digest
```

which is standard rocker shorthand, but this is also just as good.  Note that `install.packages` throws a warning and not an error if a package fails to install, so the build won't fail (e.g. will break silently).  We avoid this in `install2.r` using the `--error` or `-e` flag.

Reply:
ty! I copied this from https://irkernel.github.io/installation/. Is there a way for me to use install2.r for all the three lines (or two lines?) here?

Reply:
yup, you should only need:

```
installGithub.r IRkernel/IRkernel
```

and you can drop the `install2.r` command entirely I think.  `devtools` is already available, and I have no idea why they are telling you to install the package dependencies manually like that; since those will be installed automatically by `devtools::install_github()` since they are listed in the package DESCRIPTION: https://github.com/IRkernel/IRkernel/blob/master/DESCRIPTION#L23-L32.

To be a bit version stable, include a tag or hash in the command, e.g. probably you want:

```
installGithub.r IRkernel/IRkernel@0.8.10
```

though maybe in this case getting the latest version from master is best, i dunno.

Reply:
(you still need to keep line 12, `R --quiet -e "IRkernel::installspec()"`, since that's not part of package installation but rather using a function from the installed package)

Reply:
If I use those two lines, I instead get:

```
> IRkernel::installspec()
Error in loadNamespace(name) : there is no package called ‘IRkernel’
Calls: :: ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
Execution halted
```

I also still needed the install2.r - without it it just seems to fails to install claiming lack of deps.